src/map/map.c

```diff
@@ -2945,6 +2945,47 @@ static enum unit_dir map_calc_dir(const struct block_list *src, int16 x, int16 y
 	return dir;
 }
 
+/**
+ * Randomizes (x, y) to a walkable cell that is at least min_dist away on a square and less than max_dist.
+ *
+ * __Only__ tries each quarter of the square once, doesn't try all possibilities. (Aegis behavior)
+ * The square has a width and height of 2 * max_dist - 1.
+ * The smaller square with (x,y) center has a width and height of 2 * min_dist - 1,
+ * and is not included in the returning random cells.
+ *
+ * @remark e.g. (x + x_range, y + y_range) is not part of the square and won't be randomized.
+ * @param[in] m map ID for the walkable cell checks
+ * @param[in,out] x pointer which contains the x-axis center value of the square
+ * @param[in,out] y pointer which contains the y-axis center value of the square
+ * @param[in] min_dist minimum distance of cells from (x, y)
+ * @param[in] max_dist maximum distance of cells from (x, y)
+ * @retval 0 success, random walkable cell found
+ * @retval 1 failure, no cell found, x and y remain unchanged
+ */
+static int map_get_random_cell(int16 m, int16 *x, int16 *y, int16 min_dist, int16 max_dist)
```

Comment:
here need add bl parameter with source bl. and use it as non null in calls to this function.